### PR TITLE
solves #62: Remove all dependencies on Guava

### PR DESCRIPTION
**Pull request by royteeuwen**

### CODE REVIEW

**Comment by royteeuwen** [comment]
@kwin here is a PR. Can you please validate it from your side :)?

File: bundle/src/main/groovy/be/orbinson/aem/groovy/console/servlets/ScriptPostServlet.groovy [hunk 1]
```diff
@@ -74,16 +69,16 @@ class ScriptPostServlet extends AbstractJsonResponseServlet {
                 request: request,
                 response: response,
                 outputStream: outputStream,
-                printStream: new PrintStream(outputStream, true, Charsets.UTF_8.name()),
-                script: checkNotNull(getScript(request, scriptPath), "Script cannot be empty.")
+                printStream: new PrintStream(outputStream, true, "UTF-8"),
```
Comment (kwin) [inline bundle/src/main/groovy/be/orbinson/aem/groovy/console/servlets/ScriptPostServlet.groovy:72]: You can use https://docs.oracle.com/javase/8/docs/api/java/nio/charset/StandardCharsets.html#UTF_8 with its `name()` method

File: bundle/src/test/groovy/be/orbinson/aem/groovy/console/extension/impl/DefaultExtensionServiceTest.groovy [hunk 2]
```diff
@@ -153,7 +151,7 @@ class DefaultExtensionServiceTest {
     }
 
     private void runScriptWithExtensionService(ExtensionService extensionService) {
-        def binding = new Binding(out: new PrintStream(ByteStreams.nullOutputStream()))
+        def binding = new Binding(out: new PrintStream(OutputStream.nullOutputStream()))
```
Comment (kwin) [inline bundle/src/test/groovy/be/orbinson/aem/groovy/console/extension/impl/DefaultExtensionServiceTest.groovy:154]: I think this requires Java 11+(https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/OutputStream.html#nullOutputStream()). Probably no issue to raise requirements, but this prevents running/building with Java 8

File: bundle/src/main/groovy/be/orbinson/aem/groovy/console/servlets/ScriptPostServlet.groovy (hunk 1 above)
Comment (kwin) [reply bundle/src/main/groovy/be/orbinson/aem/groovy/console/servlets/ScriptPostServlet.groovy:72]: Java 10 also added https://docs.oracle.com/en%2Fjava%2Fjavase%2F21%2Fdocs%2Fapi%2F%2F/java.base/java/io/PrintStream.html#%3Cinit%3E(java.io.File,java.nio.charset.Charset).

File: api/src/main/groovy/be/orbinson/aem/groovy/console/api/JobProperties.groovy [hunk 3]
```diff
@@ -1,30 +1,18 @@
...
 
 @TupleConstructor
 class JobProperties {
 
-    private static final Set<String> ALL_JOB_PROPERTIES = new ImmutableSet.Builder<String>()
-            .addAll(JOB_PROPERTIES)
-            .add(DATE_CREATED)
-            .add(SCHEDULED_JOB_ID)
-            .build()
+    private static final Set<String> ALL_JOB_PROPERTIES = SetUtils.union(JOB_PROPERTIES, Set.of(DATE_CREATED, SCHEDULED_JOB_ID)).toSet()
```
Comment (kwin) [inline api/src/main/groovy/be/orbinson/aem/groovy/console/api/JobProperties.groovy:15]: Maybe use https://docs.oracle.com/javase%2F9%2Fdocs%2Fapi%2F%2F/java/util/Set.html#of-E-E-E-E-E-E- when raising requirements to Java 11

File: bundle/src/main/groovy/be/orbinson/aem/groovy/console/extension/impl/startimport/AemStarImportExtensionProvider.groovy [hunk 4]
```diff
@@ -12,7 +11,7 @@ class AemStarImportExtensionProvider implements StarImportExtensionProvider {
 
     private static final String JAVADOC_SUFFIX = "package-summary.html"
 
-    private static final Set<StarImport> IMPORTS = ImmutableSet.of(
+    private static final Set<StarImport> IMPORTS = Set.of(
```
Comment (kwin) [inline bundle/src/main/groovy/be/orbinson/aem/groovy/console/extension/impl/startimport/AemStarImportExtensionProvider.groovy:14]: This is only available in Java9+

**Comment by kwin** [comment]
You should definitely use https://maven.apache.org/plugins/maven-compiler-plugin/examples/set-compiler-release.html to check against the Java API you claim to be compatiblewith.

**Comment by royteeuwen** [comment]
> You should definitely use https://maven.apache.org/plugins/maven-compiler-plugin/examples/set-compiler-release.html to check against the Java API you claim to be compatible with.

@kwin  I spent an hour trying to get it to work, but didn't get it working. Probably the actual compiler (groovy-eclipse-compiler) does not take it correctly into account :(.

File: bundle/src/main/groovy/be/orbinson/aem/groovy/console/extension/impl/startimport/AemStarImportExtensionProvider.groovy (hunk 4 above)
Comment (royteeuwen) [reply bundle/src/main/groovy/be/orbinson/aem/groovy/console/extension/impl/startimport/AemStarImportExtensionProvider.groovy]: resolved using groovy

**Comment by royteeuwen** [comment]
Once AEM goes for a Java 17 / 21 release, I'll see if we will drop Java 8. No requirements at this moment to do so, seeing as we can use all Groovy syntax additions to add most things we want

**Comment by kwin** [comment]
> @kwin I spent an hour trying to get it to work, but didn't get it working. Probably the actual compiler (groovy-eclipse-compiler) does not take the release property correctly into account :(.

You can use https://www.mojohaus.org/animal-sniffer/animal-sniffer-maven-plugin/index.html instead as that works on the compiled classes (i.e. independent which compiler generated those)

**Comment by royteeuwen** [comment]
The animal sniffer plugin also doesn't seem to work properly with the code generated by Groovy either. Going to merge this PR as is for now

**Pull request merged**